src/GafferModule/RandomBinding.cpp

```diff
@@ -42,21 +42,80 @@
 #include "GafferBindings/DependencyNodeBinding.h"
 
 #include "Gaffer/Random.h"
+#include "Gaffer/RandomChoice.h"
 
 using namespace boost::python;
 using namespace GafferBindings;
 using namespace Gaffer;
 
-static Imath::Color3f randomColor( Random &r, int seed )
+namespace
+{
+
+Imath::Color3f randomColor( Random &r, int seed )
 {
 	return r.randomColor( std::max( seed, 0 ) );
 }
 
+void setupWrapper( RandomChoice &r, const Gaffer::ValuePlug &p )
+{
+	IECorePython::ScopedGILRelease gilRelease;
+	return r.setup( &p );
+}
+
+class RandomChoiceSerialiser : public NodeSerialiser
+{
+
+	bool childNeedsConstruction( const Gaffer::GraphComponent *child, const Serialisation &serialisation ) const override
+	{
+		const RandomChoice *node = child->parent<RandomChoice>();
+		if( child == node->outPlug() )
+		{
+			// We'll serialise a `setup()` call to construct this.
+			return false;
+		}
```

Comment:
Why do we need to do this rather than merely setting Dynamic false?

Reply:
We could set Dynamic to false, but Dynamic is going the way of the dodo, so I want to rely on it as little as possible.